jsonschema/src/keywords/required.rs

```diff
@@ -141,16 +149,27 @@ pub(crate) fn compile_with_path(
     match schema {
         Value::Array(items) => {
             if items.len() == 1 {
-                if let Some(Value::String(item)) = items.iter().next() {
+                let item = items.iter().next().unwrap();
```

Comment:
Note: I explicitly used unwrap because it must never happen, because we check len() before.

Reply:
Yep, my main motivation is that if in the future the check above will change, and a failure occur, then it will be easier to understand what is going on with `expect`

Reply:
I actually followed other approach just used index. See the changes.